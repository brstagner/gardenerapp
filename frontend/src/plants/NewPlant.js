import React from "react";
import { useState } from "react";
import TrefleApi from "../trefle/Api";
import NewPlantForm from "./NewPlantForm";
import Search from "../trefle/Search";
import "../css/forms.css";

/** Contains NewPlantForm and Search forms for adding plants to database */
function NewPlant ({ currUser }) {

    // Plant data for submission form, supplied by search or entered manually
    const [plantData, setPlantData] = useState({ id: undefined });

    // Select a cell from grid and update corresponding cell in garden object
    const handlePlantSelect = async (e) => {
        e.preventDefault();
        const id = e.target.id;
        const plant = await TrefleApi.getSpecies(id);
        const speciesData = {};
        if (plant.id) speciesData.id = plant.id;
        if (plant.common_name) speciesData.common_name = plant.common_name;
        if (plant.scientific_name) speciesData.scientific_name = plant.scientific_name;
        if (plant.flower.color) {
            speciesData.one = plant.flower.color[0] || "";
            speciesData.two = plant.flower.color[1] || "";
            speciesData.three = plant.flower.color[2] || "";
        };
        if (plant.growth.bloom_months) {
            plant.growth.bloom_months.map(month =>
                speciesData[`${month}Entry`] = true
            );
        };
        setPlantData(speciesData);

        // HIGHLIGHTING

        // Get list of potentially highlighted divs
        const plantDivs = document.querySelectorAll('.plant-div');

        // Clear highlighting
        for (let plant of plantDivs) {
            plant.classList.remove('highlighted');
        }

        // Highlight current selection
        e.target.closest('.plant-div').classList.add('highlighted');
    };

    return (
        <div>
            <h4>Add a Plant to Your Collection</h4>
            <div className="container-fluid">
                <div className="row">
                    <div className="col">
                        <Search handlePlantSelect={handlePlantSelect} />
                    </div>
                    <div className="col">
                        <NewPlantForm
                            key={plantData.id}
                            plantData={plantData}
                            currUser={currUser}
                        />
                    </div>
                </div>
            </div>
        </div>
    );
}

export default NewPlant;