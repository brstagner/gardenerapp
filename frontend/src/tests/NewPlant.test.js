import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import NewPlant from '../plants/NewPlant';

const user = { username: "user", userId: "1", isAdmin: "false", token: "user-token" };
const admin = { username: "admin", userId: "2", isAdmin: "true", token: "admin-token" };

// smoke test
it("renders without crashing", function () {
    render(
        <BrowserRouter>
            <NewPlant currUser={user} />
        </BrowserRouter>);
    // screen.debug()
});

// snapshot test
it("matches snapshot", function () {
    const { asFragment } = render(
        <BrowserRouter >
            <NewPlant currUser={user} />
        </BrowserRouter >);
    expect(asFragment()).toMatchSnapshot();
});