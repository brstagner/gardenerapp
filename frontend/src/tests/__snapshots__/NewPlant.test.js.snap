// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h4>
      Add a Plant to Your Collection
    </h4>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div>
            <h6>
              Search for a Plant
            </h6>
            <div
              class="container"
            >
              <div
                class="row"
              >
                <form
                  class=""
                  style="margin: 1em;"
                >
                  <div
                    class="col"
                  >
                    <div
                      class="mb-3 form-floating"
                    >
                      <input
                        autocomplete="name"
                        class="form-control"
                        id="name"
                        name="name"
                        placeholder="name"
                        type="text"
                        value=""
                      />
                      <label
                        class="form-label"
                        for="common_name"
                      >
                        Common or Scientific Name
                      </label>
                    </div>
                  </div>
                  <div
                    class="col"
                  >
                    <div
                      class="mb-3 form-floating"
                    >
                      <input
                        class="form-control"
                        id="color"
                        name="color"
                        placeholder="color"
                        type="text"
                      />
                      <label
                        class="form-label"
                        for="color"
                      >
                        Bloom Color
                      </label>
                    </div>
                  </div>
                  <div
                    class="col"
                  >
                    <div
                      class="mb-3"
                    >
                      Bloom Months
                      <br />
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="jan"
                        name="jan"
                        placeholder="jan"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="jan"
                      >
                        Jan
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="feb"
                        name="feb"
                        placeholder="feb"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="feb"
                      >
                        Feb
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="mar"
                        name="mar"
                        placeholder="mar"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="mar"
                      >
                        Mar
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="apr"
                        name="apr"
                        placeholder="apr"
                        type="checkbox"
                        value="false"
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="apr"
                      >
                        Apr
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="may"
                        name="may"
                        placeholder="may"
                        type="checkbox"
                        value="false"
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="may"
                      >
                        May
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="jun"
                        name="jun"
                        placeholder="jun"
                        type="checkbox"
                        value="false"
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="jun"
                      >
                        Jun
                      </label>
                      <br />
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="jul"
                        name="jul"
                        placeholder="jul"
                        type="checkbox"
                        value="false"
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="jul"
                      >
                        Jul
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="aug"
                        name="aug"
                        placeholder="aug"
                        type="checkbox"
                        value="false"
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="aug"
                      >
                        Aug
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="sep"
                        name="sep"
                        placeholder="sep"
                        type="checkbox"
                        value="false"
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="sep"
                      >
                        Sep
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="oct"
                        name="oct"
                        placeholder="oct"
                        type="checkbox"
                        value="false"
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="oct"
                      >
                        Oct
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="nov"
                        name="nov"
                        placeholder="nov"
                        type="checkbox"
                        value="false"
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="nov"
                      >
                        Nov
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="dec"
                        name="dec"
                        placeholder="dec"
                        type="checkbox"
                        value="false"
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="dec"
                      >
                        Dec
                      </label>
                    </div>
                  </div>
                  <button
                    class="btn btn-secondary"
                  >
                    Search
                  </button>
                </form>
              </div>
              <div
                class="row"
              >
                <div />
              </div>
            </div>
          </div>
        </div>
        <div
          class="col"
        >
          <div>
            <h6>
              Enter plant information
            </h6>
            <div
              class="container"
            >
              <div
                class="row"
              >
                <form
                  class=""
                  style="margin: 1em;"
                >
                  <div
                    class="col"
                  >
                    <div
                      class="mb-3 form-floating"
                    >
                      <input
                        autocomplete="common_name"
                        class="form-control"
                        id="common_name"
                        name="common_name"
                        placeholder="common_name"
                        type="text"
                        value=""
                      />
                      <label
                        class="form-label"
                        for="common_name"
                      >
                        Common Name
                      </label>
                    </div>
                  </div>
                  <div
                    class="col"
                  >
                    <div
                      class="mb-3 form-floating"
                    >
                      <input
                        autocomplete="scientific_name"
                        class="form-control"
                        id="scientific_name"
                        name="scientific_name"
                        placeholder="scientific_name"
                        type="text"
                        value=""
                      />
                      <label
                        class="form-label"
                        for="scientific_name"
                      >
                        Scientific Name
                      </label>
                    </div>
                  </div>
                  <div
                    class="col"
                  >
                    Bloom Colors
                    <div
                      class="mb-3 form-floating"
                    >
                      <input
                        class="form-control"
                        id="one"
                        name="one"
                        placeholder="one"
                        type="text"
                        value=""
                      />
                      <label
                        class="form-label"
                        for="one"
                      >
                        Bloom Color One
                      </label>
                    </div>
                    <div
                      class="mb-3 form-floating"
                    >
                      <input
                        class="form-control"
                        id="two"
                        name="two"
                        placeholder="two"
                        type="text"
                        value=""
                      />
                      <label
                        class="form-label"
                        for="two"
                      >
                        Bloom Color Two
                      </label>
                    </div>
                    <div
                      class="mb-3 form-floating"
                    >
                      <input
                        class="form-control"
                        id="three"
                        name="three"
                        placeholder="three"
                        type="text"
                        value=""
                      />
                      <label
                        class="form-label"
                        for="three"
                      >
                        Bloom Color Three
                      </label>
                    </div>
                  </div>
                  <div
                    class="col"
                  >
                    <div
                      class="mb-3"
                    >
                       Bloom Months
                      <br />
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="janEntry"
                        name="janEntry"
                        placeholder="jan"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="janEntry"
                      >
                        Jan
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="febEntry"
                        name="febEntry"
                        placeholder="feb"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="febEntry"
                      >
                        Feb
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="marEntry"
                        name="marEntry"
                        placeholder="mar"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="marEntry"
                      >
                        Mar
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="aprEntry"
                        name="aprEntry"
                        placeholder="apr"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="aprEntry"
                      >
                        Apr
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="mayEntry"
                        name="mayEntry"
                        placeholder="may"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="mayEntry"
                      >
                        May
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="junEntry"
                        name="junEntry"
                        placeholder="jun"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="junEntry"
                      >
                        Jun
                      </label>
                      <br />
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="julEntry"
                        name="julEntry"
                        placeholder="jul"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="julEntry"
                      >
                        Jul
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="augEntry"
                        name="augEntry"
                        placeholder="aug"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="augEntry"
                      >
                        Aug
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="sepEntry"
                        name="sepEntry"
                        placeholder="sep"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="sepEntry"
                      >
                        Sep
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="octEntry"
                        name="octEntry"
                        placeholder="oct"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="octEntry"
                      >
                        Oct
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="novEntry"
                        name="novEntry"
                        placeholder="nov"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="novEntry"
                      >
                        Nov
                      </label>
                      <input
                        autocomplete="off"
                        class="btn-check form-check-input"
                        id="decEntry"
                        name="decEntry"
                        placeholder="dec"
                        type="checkbox"
                        value=""
                      />
                      <label
                        class="btn btn-outline-primary form-label"
                        for="decEntry"
                      >
                        Dec
                      </label>
                    </div>
                  </div>
                  <button
                    class="btn btn-secondary"
                  >
                    Add Plant
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
